kotlinpoet/src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt

```diff
@@ -206,13 +206,13 @@ class CodeBlockTest {
     val typeBlock = CodeBlock.of("%T", type)
     assertThat(typeBlock.toString()).isEqualTo("kotlin.String?")
 
-    val list = List::class.asClassName().copy(nullable = true)
+    val list = (List::class.asClassName().copy(nullable = true) as ClassName)
```

Comment:
We should fix this to return self-types instead of `TypeName`, this is pretty awkward to use at the moment.

Reply:
Yep, a v2 could be something like

```kotlin
sealed class TypeName<T : TypeName> {
  abstract fun copy(...): T
}

class ClassName : TypeName<ClassName> {

}
```